modules/navier_stokes/include/utils/Reconstructions.h

```diff
@@ -55,7 +55,13 @@ interpolateReconstruct(CellCenteredMapFunctor<T, Map> & output_functor,
     mooseAssert(face, "This must be non-null");
     const Real weight = weight_with_sf ? face->faceArea() * face->faceCoord() : 1;
     const Moose::FaceArg face_arg{
-        face, Moose::FV::LimiterType::CentralDifference, true, false, nullptr};
+        face,
+        Moose::FV::LimiterType::CentralDifference,
+        true,
+        false,
+        input_functor.hasFaceSide(*face, true)
+            ? (input_functor.hasFaceSide(*face, false) ? nullptr : face->elemPtr())
+            : face->neighborPtr()};
```

Comment:
Yea the idea behind the `checkFace` originally was to offload work for the downstream developer, but if we're seeing large gains in performance, I support the change

Reply:
On the same problem (3D, 500k cells, 1 processor) this change reduced the computation time from 137 s to 131 s on average which is ~4.4%.